test/performance/odp_ipsec.c

```diff
@@ -726,52 +760,63 @@ run_measure_one_async(ipsec_args_t *cargs,
 	       (packets_received < cargs->iteration_count)) {
 		odp_event_t ev;
 
-		if ((packets_sent < cargs->iteration_count) &&
+		num_pkts = cargs->iteration_count - packets_sent;
+		num_pkts = num_pkts > burst_sz ? burst_sz : num_pkts;
+
+		if (num_pkts &&
 		    (packets_sent - packets_received <
 		     cargs->in_flight)) {
-			pkt = make_packet(pkt_pool, payload_length);
-			if (ODP_PACKET_INVALID == pkt)
+			if (odp_unlikely(make_packet_multi(pkt_pool,
+							   payload_length,
+							   pkt,
+							   num_pkts)))
 				return -1;
 
-			if (cargs->debug_packets)
-				odp_packet_print_data(pkt, 0,
-						      odp_packet_len(pkt));
+			if (cargs->debug_packets) {
+				for (i = 0; i < num_pkts; i++)
+					odp_packet_print_data(pkt[i], 0,
+							      odp_packet_len(pkt[i]));
+			}
 
-			rc = odp_ipsec_out_enq(&pkt, 1,
+			rc = odp_ipsec_out_enq(pkt, num_pkts,
 					       &param);
 			if (rc <= 0) {
 				ODPH_ERR("failed odp_crypto_packet_op_enq: rc = %d\n",
 					 rc);
-				odp_packet_free(pkt);
+				odp_packet_free_sp(pkt, num_pkts);
 				break;
 			}
 			packets_sent += rc;
 		}
 
+		 i = 0;
+
 		if (cargs->schedule)
 			ev = odp_schedule(NULL,
 					  ODP_SCHED_NO_WAIT);
 		else
 			ev = odp_queue_deq(out_queue);
 
 		while (ev != ODP_EVENT_INVALID) {
-			odp_packet_t out_pkt;
```

Comment:
Maybe this async test loop should use odp_schedule_multi() now?

Reply:
PR 1506 takes odp_schedule_multi and odp_queue_deq_multi into use here. Please take a look.